add the private flag for mount the rootfs with different flag when run a container.

cc272309126 opened the pull request:
1.the priavte flag is default true, which means mount the rootfs with --make-private.
2.if specify the private flag false, which means mount the rootfs with --make-slave.

Signed-off-by: Chao Chen chenchao@qiyi.com

phemmer commented:
There is already a PR up with more flexible solution which also implements the feature in this PR: #15648

cpuguy83 commented:
Thanks for submitting! Closing this one in favor of the other that @phemmer mentioned.

Commit:
add the private flag for mount the rootfs with different flag when run a container.
1.the priavte flag is default true, which means mount the rootfs with --make-private.
2.if specify the private flag false, which means mount the rootfs with --make-slave.

Signed-off-by: Chao Chen <chenchao@qiyi.com>
